src/cmds/service.c

```diff
@@ -5,21 +5,93 @@
  * @date 16.04.16
  * @author Alexander Kalmuk
  */
-
+#include <util/log.h>
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <mem/misc/pool.h>
+#include <util/dlist.h>
+#include <sys/types.h>
+#include <signal.h>
+
+#define SIZE_OF_SERVICE_NAME 32
+#define OBJECTS_QUANTITY 0x10
+
+struct service_list {
+	struct dlist_head dlist_item;
+	char service_name[SIZE_OF_SERVICE_NAME];
+	pid_t pid;
+};
+
+POOL_DEF(pool, struct service_list, OBJECTS_QUANTITY);		 
+
+static DLIST_DEFINE(head);
+
+static void service_add(const char *name, pid_t service_pid) {
+  	struct service_list *new_element;
+	struct service_list *tmp;
+		
+	dlist_foreach_entry(tmp, &head, dlist_item) {
+		if (strcmp(tmp->service_name, name) == 0) {
+			printf("Service is already running.\n");
+			return;
+		}
+	}
+
+  	new_element = pool_alloc(&pool);
+
+	if (new_element == NULL) {
+		log_error("Pool is full.");
+		return;
+	}
+	
+  	strncpy(new_element->service_name,name,SIZE_OF_SERVICE_NAME);
+
+	new_element->pid = service_pid;
+   
+  	dlist_init(&new_element->dlist_item);
+  
+  	dlist_add_next(&new_element->dlist_item, &head); 
+}
 
-static void srv_print_services(void) {
-	printf("PRINT LIST OF SERVICES IS NOT SUPPORTED YET\n");
+static void service_delete(const char *name) {
+	struct service_list *tmp;
+	
+	dlist_foreach_entry(tmp, &head, dlist_item) {
+		if (strcmp(tmp->service_name, name) == 0) {
+			int kill_result = kill(tmp->pid, SIGKILL);
+			
+			if(kill_result != 0) {
+				printf("Error: the service was launched but is not working now.\n");
+				return;
```

Comment:
If service task is not working, it's still necessary to remove it from the service list.